lms/djangoapps/teams/tests/test_views.py

```diff
@@ -1708,3 +1709,60 @@ def test_download_csv(self):
     def get_url(course_id):
         # This strategy allows us to test with invalid course IDs
         return reverse('team_membership_bulk_management', args=[course_id])
+
+    def test_upload_valid_csv_simple(self):
+        self.create_and_enroll_student(username='a_user')
+        csv_content = 'user,mode,topic_0' + '\n'
+        csv_content += 'a_user, masters, team wind power'
+        csv_file = SimpleUploadedFile('test_file.csv', csv_content.encode('utf8'), content_type='text/csv')
+        self.client.login(username=self.users['course_staff'].username, password=self.users['course_staff'].password)
+        self.make_call(reverse('team_membership_bulk_management', args=[self.good_course_id]),
+                       201, method='post',
+                       data={'csv': csv_file}, user='staff'
+                       )
+
+    def test_upload_invalid_teamset(self):
+        self.create_and_enroll_student(username='a_user')
+        csv_content = 'user,mode,topic_0_bad' + '\n'
+        csv_content += 'a_user, masters, team wind power'
+        csv_file = SimpleUploadedFile('test_file.csv', csv_content.encode('utf8'), content_type='text/csv')
+        self.client.login(username=self.users['course_staff'].username, password=self.users['course_staff'].password)
+        self.make_call(reverse('team_membership_bulk_management', args=[self.good_course_id]),
+                       400, method='post',
+                       data={'csv': csv_file}, user='staff'
+                       )
+
+    def test_upload_assign_user_twice_to_same_teamset(self):
+        csv_content = 'user,mode,topic_0' + '\n'
+        csv_content += 'student_enrolled, masters, team wind power'
+        csv_file = SimpleUploadedFile('test_file.csv', csv_content.encode('utf8'), content_type='text/csv')
+        self.client.login(username=self.users['course_staff'].username, password=self.users['course_staff'].password)
+        self.make_call(reverse('team_membership_bulk_management', args=[self.good_course_id]),
+                       400, method='post',
+                       data={'csv': csv_file}, user='staff'
+                       )
+
+    def test_upload_assign_one_user_to_different_teamsets(self):
+        self.create_and_enroll_student(username='a_user')
+        self.create_and_enroll_student(username='b_user')
+        self.create_and_enroll_student(username='c_user')
+        csv_content = 'user,mode,topic_0,topic_1,topic_2' + '\n'
```

Comment:
Why not use a different test file so you don't have to write this in code?

Reply:
Can you elaborate on that? This is needed for a specific scenario in a single test.

Reply:
NVM, I thought it was loading a test asset, but instead all these tests are in-code.